libevmasm/AssemblyItem.h

```diff
@@ -263,12 +290,27 @@ class AssemblyItem
 
 	void setImmutableOccurrences(size_t _n) const { m_immutableOccurrences = _n; }
 
+	struct FunctionSignature
+	{
+		uint8_t argsNum;
+		uint8_t retsNum;
+		bool canContinue;
+	};
```

Comment:
I see that you treat `retsNum == 0x80` as a special case everywhere, which makes `canContinue` redundant. IMO it would be better to make it a function that checks for `0x80`. Or you could store `0` in `retsNum` when `canContinue` is `false`, which would save you some checks for this special case.

BTW, this should all be documented in a docstring. I.e. what the allowed ranges of the fields are and how `retsNum` behaves behaves when `canContinue == false`.

Reply:
Yes. That was my idea too but didn't want to change to much. But if you pointed it out too I will change accordingly.

Reply:
> I would rather go with the first option because `retsNum` can be 0 and function can continue a t the same time.

What I meant was to just always rely on the flag and have `retsNum` literally reprsent the number of returned values. `retsNum == 0` would be possible with either value of `canContinue`. But anyway, either solution is fine.

Reply:
One more very important thing though. I think that `m_functionSignature` should be included in comparison performed by `==` and `<` operators. Otherwise we'll get another deduplicator bug like #14649.

Reply:
> I think that `m_functionSignature` should be included in comparison performed by `==` and `<` operators.

Actually it's not necessary. Now that I think of it, we have function ID in the `data()`, which is already a part of the comparison and that uniquely determines the signature.